drivers/serial/uart_stm32.c

```diff
@@ -395,6 +395,33 @@ static void uart_stm32_poll_out(struct device *dev,
 	LL_USART_TransmitData8(UartInstance, (u8_t)c);
 }
 
+static int uart_stm32_err_check(struct device *dev)
+{
+	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+	int err = 0;
+
+	if (LL_USART_IsActiveFlag_ORE(UartInstance)) {
+		LL_USART_ClearFlag_ORE(UartInstance);
+		err |= UART_ERROR_OVERRUN;
+	}
+
+	if (LL_USART_IsActiveFlag_PE(UartInstance)) {
+		LL_USART_ClearFlag_PE(UartInstance);
+		err |= UART_ERROR_PARITY;
+	}
+
+	if (LL_USART_IsActiveFlag_FE(UartInstance)) {
+		LL_USART_ClearFlag_FE(UartInstance);
+		err |= UART_ERROR_FRAMING;
+	}
+
+	if (LL_USART_IsActiveFlag_NE(UartInstance)) {
```

Comment:
Please add a comment explaining why this is done here

Reply:
I removed it completely as enum doesn't provide such error definition and it seems to be vendor specific. Furthermore, I don't see any side effects (like in ore) to leave it set if it is raised.